src/probnum/statespace/discrete_transition.py

```diff
@@ -14,17 +15,8 @@
 
 try:
     # functools.cached_property is only available in Python >=3.8
-    from functools import cached_property, lru_cache
+    from functools import cached_property
 except ImportError:
-    from functools import lru_cache
-
-    from cached_property import cached_property
-
-try:
-    # functools.cached_property is only available in Python >=3.8
-    from functools import cached_property, lru_cache
-except ImportError:
-    from functools import lru_cache
 
```

Comment:
I tidied some stuff up while I was at it. This is not related to the rest of the PR.

Reply:
if these are not used, then great!